### PR TITLE
ROX-21064: Deprecate custom SCCs in Operator [WIP]

### PR DESCRIPTION
**Pull request by Maddosaurus**
## Description

This is part 2 of a PR stack, focusing exclusively on the needed changes for the Operator and its tests.

* **PR #8663** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8663?utm_source=stack-comment-icon" target="_blank"></a> 
* **PR #8766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8766?utm_source=stack-comment-icon" target="_blank"></a> 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by Maddosaurus** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8663** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8663?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8766?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/8766?utm_source=stack-comment).

File: operator/bundle/manifests/platform.stackrox.io_centrals.yaml [hunk 1]
```diff
@@ -686,10 +686,14 @@ spec:
                   type: object
                 type: array
               misc:
-                description: Miscellaneous settings.
+                description: Deprecated field. It is no longer necessary to specify
+                  it. This field will be removed in a future release. Miscellaneous
+                  settings.
```
Comment (vikin91) [inline operator/bundle/manifests/platform.stackrox.io_centrals.yaml:691]: It reads a bit weird. How about:
```suggestion
                description: Deprecated field. This field will be removed in a future release. Miscellaneous
                  settings.
```

**Review by vikin91 — COMMENTED** [review]
Please take into consideration comments of @ludydoo from https://github.com/stackrox/stackrox/pull/7671

File: operator/bundle/manifests/platform.stackrox.io_centrals.yaml (hunk 1 above)
Comment (Maddosaurus) [reply operator/bundle/manifests/platform.stackrox.io_centrals.yaml]: The manifests are generated. I've updated the generation source to rephrase the message in https://github.com/stackrox/stackrox/pull/8766/commits/b10734963574a9ca248cb38bc0bedfcad52522d4

File: operator/apis/platform/v1alpha1/central_types.go [hunk 2]
```diff
@@ -53,16 +53,17 @@ type CentralSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName=Customizations,order=6,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
 	Customize *CustomizeSpec `json:"customize,omitempty"`
 
+	// Deprecated field. This field will be removed in a future release.
```
Comment (porridge) [inline operator/apis/platform/v1alpha1/central_types.go:56]:
```suggestion
	// Deprecated field. This field will be removed in a future API version.
```

I don't think we can ever remove fields from an existing API version...

**Comment by Maddosaurus** [comment]
Superseded by https://github.com/stackrox/stackrox/pull/8781

**Pull request closed without merging**